core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -0,0 +1,910 @@
+      catalog().createNamespace(identifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+
+    View view =
+        catalog()
+            .buildView(identifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(identifier.namespace())
+            .withQuery("spark", "select * from ns.tbl")
+            .create();
+
+    assertThat(view).isNotNull();
+    assertThat(catalog().viewExists(identifier)).as("View should exist").isTrue();
+
+    assertThatThrownBy(
+            () ->
+                catalog()
+                    .buildView(identifier)
+                    .withSchema(OTHER_SCHEMA)
+                    .withQuery("spark", "select * from ns.tbl")
+                    .withDefaultNamespace(identifier.namespace())
+                    .create())
+        .isInstanceOf(AlreadyExistsException.class)
+        .hasMessageStartingWith("View already exists: ns.view");
+  }
+
+  @Test
+  public void createViewThatAlreadyExistsAsTable() {
+    Assumptions.assumeThat(tableCatalog())
+        .as("Only valid for catalogs that support tables")
+        .isNotNull();
+
+    TableIdentifier tableIdentifier = TableIdentifier.of("ns", "table");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(tableIdentifier.namespace());
+    }
+
+    assertThat(tableCatalog().tableExists(tableIdentifier)).as("Table should not exist").isFalse();
+
+    tableCatalog().buildTable(tableIdentifier, SCHEMA).create();
+
+    assertThat(tableCatalog().tableExists(tableIdentifier)).as("Table should exist").isTrue();
+
+    assertThatThrownBy(
+            () ->
+                catalog()
+                    .buildView(tableIdentifier)
+                    .withSchema(OTHER_SCHEMA)
+                    .withDefaultNamespace(tableIdentifier.namespace())
+                    .withQuery("spark", "select * from ns.tbl")
+                    .create())
+        .isInstanceOf(AlreadyExistsException.class)
+        .hasMessageStartingWith("Table with same name already exists: ns.table");
+  }
+
+  @Test
+  public void createTableThatAlreadyExistsAsView() {
+    Assumptions.assumeThat(tableCatalog())
+        .as("Only valid for catalogs that support tables")
+        .isNotNull();
+
+    TableIdentifier viewIdentifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(viewIdentifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(viewIdentifier)).as("View should not exist").isFalse();
+
+    catalog()
+        .buildView(viewIdentifier)
+        .withSchema(SCHEMA)
+        .withDefaultNamespace(viewIdentifier.namespace())
+        .withQuery("spark", "select * from ns.tbl")
+        .create();
+
+    assertThat(catalog().viewExists(viewIdentifier)).as("View should exist").isTrue();
+
+    assertThatThrownBy(() -> tableCatalog().buildTable(viewIdentifier, SCHEMA).create())
```

Comment:
I think we also need tests for for the behavior of `buildTable(...).replaceTransaction()` and similar, including the view methods (`buildView(...).replace()`). I think that the behavior should be that the replace fails because it would change the type of object in the catalog, rather than replacing a view with a table or vice versa.

Reply:
I've addressed this in 76a3f606a1e13faa8065c19400ddbe35460b30fd. Let me know if that's what you had in mind